### ISSUE TITLE
Add missing return codes for security manager

### ISSUE THREAD

**Issue by sampaioletti** [taks/esp32-nimble#93]
Add missing return codes for security manager to `BLEReturnCode` https://github.com/taks/esp32-nimble/blob/8f55ab516bc2854ac7304ff7875e8aa0066546f0/src/ble_return_code.rs#L39

[https://mynewt.apache.org/latest/network/ble_hs/ble_hs_return_codes.html#return-codes-security-manager-us](https://mynewt.apache.org/latest/network/ble_hs/ble_hs_return_codes.html#return-codes-security-manager-us)

As a secondary point might be useful to consider returning `BLEReturnCode` in some additional places. i.e.

https://github.com/taks/esp32-nimble/blob/8f55ab516bc2854ac7304ff7875e8aa0066546f0/src/server/ble_server.rs#L89

instead of c_int, which then begs the question should `BLEReturnCode` inner be a `i32` or is `u32` correct.

https://github.com/taks/esp32-nimble/blob/8f55ab516bc2854ac7304ff7875e8aa0066546f0/src/ble_return_code.rs#L2

**Comment by taks** [comment]
Can you create a PR?

**Comment by sampaioletti** [comment]
Yeah I should have some time this week. I'll try and get something put together.


-Sam
________________________________

**Comment by sampaioletti** [comment]
What is the reasoning behind the wrapped type for `BLEReturnCode` vs a more rusty Enum style error? Not a critique, just an ask.

**Comment by taks** [comment]
At the time of implementation,
if I was not sure whether the value could be converted to a `BLEReturnCode`, I used the value as it is.

It is better to replace `BLEReturnCode` if it can be replaced.
